### PR TITLE
Improved diff to markdown for card notifications

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
The diff libraries available for Go are designed for source code and patches, meaning they are line based or character based.  This PR improves the diff output by working with words as much as possible and better handling large text changes.  This PR also provides better detection of superficial changes that do not require a notification (e.g. adding/removing whitespace).

#### Ticket Link
https://community.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/b4ctmm917q3rzdfi451qtuxjzcw/150d1464-6d59-4c39-810f-ee201439dbaf/caakrwrbf9tr6bgutoyok8ky1ir

### CODE REVIEW

File: server/services/notify/notifysubscriptions/diff2markdown.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+import (
+	"strings"
+
+	"github.com/sergi/go-diff/diffmatchpatch"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+func diff2Markdown(oldText string, newText string, logger *mlog.Logger) string {
```
Comment (jespino) [inline server/services/notify/notifysubscriptions/diff2markdown.go:11]: I don't know if the name is a bit misleading here, I would expect the function with this name to get a "diff" and generate a "markdown", not to generate a diff  and return it in markdown. Maybe I would name it as `generateMarkdownDiff` or `getMarkdownDiff` or something like that.

File: server/services/notify/notifysubscriptions/diff2markdown_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package notifysubscriptions
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_reverse(t *testing.T) {
```
Comment (jespino) [inline server/services/notify/notifysubscriptions/diff2markdown_test.go:9]: Maybe some other tests would be helpful, things like the truncation function or the markdown generation from the diffs

**Review by jespino — CHANGES_REQUESTED** [review]
In general looks good to me, added to suggestions to improve that.

File: server/services/notify/notifysubscriptions/diff2markdown_test.go (hunk 2 above)
Comment (wiggin77) [reply server/services/notify/notifysubscriptions/diff2markdown_test.go:9]: I created ticket https://community.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/b4ctmm917q3rzdfi451qtuxjzcw/150d1464-6d59-4c39-810f-ee201439dbaf/c4p1m99neap8zdm59gb1u7361zw to add additional unit tests.  I'll get to this very soon, but I don't want to delay merge waiting for it.

File: server/services/notify/notifysubscriptions/diff2markdown.go (hunk 1 above)
Comment (wiggin77) [reply server/services/notify/notifysubscriptions/diff2markdown.go]: Fixed via https://github.com/mattermost/focalboard/pull/2037/commits/39420ad6d53efe4e7377afa5b197b8fdfbcf3526

**Review by jespino — APPROVED** [review]
LGTM

**Review by sbishel — APPROVED** [review]

**Pull request merged**